.claude/skills/cookbook-audit/validate_notebook.py

```diff
@@ -0,0 +1,386 @@
+            return
+
+        intro_text = self.get_cell_source(first_cell)
+
+        # Check for minimum length
+        if len(intro_text) < 200:
+            self.warnings.append(
+                "Introduction seems too short (< 200 chars). Should include "
+                "problem statement, audience, and what will be accomplished"
+            )
+
+        # Check for key elements
+        has_prerequisites = bool(re.search(r'prerequisite|requirement|need|require', intro_text, re.IGNORECASE))
+        if not has_prerequisites:
+            self.warnings.append("Introduction doesn't mention prerequisites")
+
+    def check_pip_install_output(self):
+        """Check that pip install outputs are suppressed."""
+        for i, cell in enumerate(self.cells):
+            if cell.get('cell_type') != 'code':
+                continue
+
+            source = self.get_cell_source(cell)
+            has_pip_install = 'pip install' in source
+            has_capture = '%%capture' in source or '%pip install' in source
+
+            if has_pip_install and not has_capture:
+                self.warnings.append(
+                    f"Cell {i}: pip install without output suppression "
+                    "(use %%capture or %pip)"
+                )
+
+    def check_code_explanations(self):
+        """Check that code blocks have explanatory text before them."""
+        prev_cell_type = None
+
+        for i, cell in enumerate(self.cells):
+            cell_type = cell.get('cell_type')
+
+            if cell_type == 'code' and prev_cell_type == 'code':
+                # Two code cells in a row - might be missing explanation
+                source = self.get_cell_source(cell)
+                # Skip if it's just a simple continuation (e.g., print statement)
+                if source.strip() and not source.strip().startswith('#'):
+                    self.warnings.append(
+                        f"Cell {i}: Code cell without preceding markdown "
+                        "explanation (two code cells in a row)"
+                    )
+
+            prev_cell_type = cell_type
+
+    def check_verbose_output(self):
+        """Check for verbose debug output."""
+        verbose_patterns = [
+            r'print\(["\']debug',
+            r'\.debug\(',
+            r'verbose\s*=\s*True',
+        ]
+
+        for i, cell in enumerate(self.cells):
+            if cell.get('cell_type') != 'code':
+                continue
+
+            source = self.get_cell_source(cell)
+            for pattern in verbose_patterns:
+                if re.search(pattern, source, re.IGNORECASE):
+                    self.warnings.append(
+                        f"Cell {i}: Contains verbose debug output"
+                    )
+
+    def check_variable_names(self):
+        """Check for poor variable naming."""
+        poor_names = [r'\bx\d*\b', r'\btemp\d*\b', r'\bresult\d*\b', r'\bdata\d*\b']
```

Comment:
may we use some sort of linter for this... for example https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html allows specification of regexes like you're doing here.

Reply:
or better yet [black](https://black.readthedocs.io/en/stable/getting_started.html) via `pip install "black[jupyter]"` since we use black commonly at Ant.

Reply:
same as above, will move to a new PR